### ISSUE TITLE
Music library fails to detect new songs

### ISSUE THREAD

**Issue by motatuc** [vanilla-music/vanilla#533]
Hello

I am facing a bug with the music library.

### Bug description
The music library fails to detect new songs. After copying a new album to the phone, it does not show up in vanillla. It does not get detected automatically and also manually using Start Scan and/or Scan full filesystem does not help either.

The album does not show up. It has correct metatags, is mp3 format, and androids own mp3 player does find it.

### Steps to reproduce
- Having already a library of 1000 songs
- Add another album
- Wait or run the library scan manually

**Actual result:** The album is not detected by vanilla
**Expected result:** The album should be detected (automatically and at least manually)

### Device info
**Android version:** Lineage 14.1
**Vanilla version:** 1.48

### Additional information:
The music library is failing to detect new songs, no matter what album/song/format is added. My guess is that one particular song makes problems and kills the library scan. Because if I flush the library and let it start from scratch all the songs got recognized.

### Possible problem see log:
>02-02 21:49:29.735   274   418 I FFmpegExtractor: android-source:0xea7e5200
02-02 21:49:29.738   274   418 D FFmpegExtractor: unsuppoted codec(flac) by official Stagefright
02-02 21:49:29.738   274   418 D FFmpegExtractor: ffmpeg detected media content as 'audio/flac' with confidence 0.88
02-02 21:49:29.744   274   418 I FFmpegExtractor: the duration is 00:03:07.02
02-02 21:49:29.744   274   418 I FFmpegExtractor: stream_index: 0
02-02 21:49:29.744   274   418 I FFmpegExtractor: support the codec(flac) disposition(0)
02-02 21:49:29.744   274   418 I hexdump : 00000000:  10 00 10 00 00 00 0e 00  2f ed 0a c4 42 f0 00 7d  ......../...B..}
02-02 21:49:29.744   274   418 I hexdump : 00000010:  da 44 0d db ea 7c e6 05  2e ac a2 0f 3f c1 28 45  .D...|......?.(E
02-02 21:49:29.744   274   418 I hexdump : 00000020:  4a 47                                             JG
02-02 21:49:29.744   274   418 D FFmpegExtractor: bit_rate: 0, sample_rate: 44100, channels: 2, bits_per_coded_sample: 0, block_align: 0 bits_per_raw_sample: 16, sample_format: 1
02-02 21:49:29.744   274   418 I FFmpegExtractor: the time is 00:03:07.02
02-02 21:49:29.744   274   418 D FFmpegExtractor: Reader thread starting
02-02 21:49:29.744   274   418 D FFmpegExtractor: supported mime: audio/flac
02-02 21:49:29.744   274   418 I MediaExtractor: Use extended extractor for the special mime(audio/flac) or codec
02-02 21:49:29.762   274   274 D FFmpegExtractor: Reader thread stopped

**Comment by adrian-bl** [comment]
> Because if I flush the library and let it start from scratch all the songs got recognized.

That is odd: So you say that checking the 'flush library' button is required to get the library populated?
A 'simple' full-rescan doesn't help?

The output of 

 'logcat | grep VanillaMusic' 

might give some clues as it displays some information about the scan process.

**Comment by motatuc** [comment]
Yes, flush library is required to get the library populated completely.
Right, a 'simple' scan does not help.

The scan process always seems to stop at the same file. Could it be a problem of the files name?

>MediaScanner: inserted /storage/Hyun-jung Shim Old Boy/09 Vivaldi ''Four Seasons'' Concerto RV.297 (Perf. By Saint Luke Chamber Orchestra).flac

**Comment by motatuc** [comment]
Well,

I renamed this file, did a new scan (quick scan and also full system scan) and now I got one song less than before! Interesting...

**Comment by motatuc** [comment]
I saw that you increased logging so I gave it another shot. 

Still, the album is not added and the scan stops with this last file again.
Also I see output like this:

>Error calling setDataSource for /storage/emulated/0/Download/vanilla-debug.apk: java.lang.RuntimeException: setDataSource failed: status = 0x80000000

This happens for some other files with other file endings as well...

**Comment by adrian-bl** [comment]
> MediaScanner: inserted /storage/My own music album/01 Song ''Four Seasons'' Intime AB.397 
> (Perf. By Saint Martin Classic Orchestra).flac

That is the last file which was *successfully* indexed, so the issue is most likely caused by the next file


> I saw that you increased logging so I gave it another shot.

Yes: You should now see something like 'Extracting tags from' ....
This is logged before we even start to open a file, so if the scan process gets stuck / abort, then the file logged there is most likely to blame.

Also: Do you see the scan progress finishing? Eg: Does the scanning-notification ever disappear?

> Error calling setDataSource for /storage/emulated/0/Download/vanilla-debug.apk: 
> java.lang.RuntimeException: setDataSource failed: status = 0x80000000

That is 'normal' for non-music files: It just means that androids tagreader does not think that this is a playable file (which is true in this case)

**Comment by motatuc** [comment]
> That is the last file which was successfully indexed, so the issue is most likely caused by the next file

Which I unfortunately even with the increased logging are not able to find out.

>Also: Do you see the scan progress finishing? Eg: Does the scanning-notification ever disappear?

Yes, the notification disappears.

Now, with the new option to use the internal tag-editor the scan seems to find all the available sound files (found 20 something more).

I'll stick to that option, maybe you should make it default.

**Comment by adrian-bl** [comment]
> Now, with the new option to use the internal tag-editor the scan seems to find all the available sound 
> files (found 20 something more).

Ok, so at least that works - but the behavior of your device is still very strange.

The internal tagreader is already the default for Flac, Ogg and Opus files - but i'm not fully confident in my ID3v2 parser - but if enough people get good results out of it, then i really consider to use it as a default. (The ID3-Tag specification is a *mess*)

**Comment by adrian-bl** [comment]
Closing this bug as 1.0.49 is about to be released

**Comment by maples7009** [comment]
I came across this thread with a similar issue.  I had recently added some songs to my Music folder, but Vanilla wasn't showing the new songs.  None of the options on the "Media Scan" screen worked.  
After uninstallig and re-installing, I finally realized that at some point in the past, Vanilla must have ignored the ".nomedia" tags (instead, just directly followed the folders set in the "Indexed Directories") but then started to follow them, ignoring anything with a .nomedia file.  For some reason, I had a .nomedia file in my Music directory, so Vanilla was not scanning it anymore.  Removing the .nomedia file solved the problem.

Don't ask me why I had a .nomedia file in my Music folder, it doesn't make sense to me either :)
